metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -104,110 +123,174 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+	vmChunks := m.getVmChunks(vmList)
+	vmMap := getVmMap(vmList)
+
+	for _, chunk := range vmChunks {
+		// Perform the query for all VMs at once
+		metricBase, err := perfManager.SampleByName(ctx, querySpec, metricsToQuery, chunk)
+		if err != nil {
+			fmt.Printf("Error retrieving metrics: %s\n", err)
+			return nil // this probably needs to be a different error
 		}
 
-		if vm.Summary.Vm != nil {
-			networkNames, err := getNetworkNames(ctx, c, vm.Summary.Vm.Reference())
-			if err != nil {
-				m.Logger().Debug(err.Error())
-			} else {
-				if len(networkNames) > 0 {
-					event["network_names"] = networkNames
+		// assign metrics to the event output
+		for _, base := range metricBase {
+			if metric, ok := base.(*types.PerfEntityMetric); ok {
+
+				// build initial even tobject with static values
+				virtualMachine := vmMap[metric.Entity.Value]
+				event := mapstr.M{
+					"name": virtualMachine.Summary.Config.Name,
+					"os":   virtualMachine.Summary.Config.GuestFullName,
+					"uuid": virtualMachine.Summary.Config.InstanceUuid,
+					"id":   virtualMachine.Reference().Value,
+					"resource_pool": mapstr.M{
+						"id": virtualMachine.ResourcePool.Value,
+					},
+					"cpu": mapstr.M{
+						"reserved": virtualMachine.Summary.Config.CpuReservation,
+						"cores":    virtualMachine.Summary.Config.NumCpu,
+					},
+					"disks": mapstr.M{
+						"count": virtualMachine.Summary.Config.NumVirtualDisks,
+					},
+					"storage": mapstr.M{
+						"committed":  virtualMachine.Summary.Storage.Committed,
+						"uncommited": virtualMachine.Summary.Storage.Uncommitted,
+						"total":      virtualMachine.Summary.Storage.Committed + virtualMachine.Summary.Storage.Uncommitted,
+					},
+					"heartbeat_status": virtualMachine.GuestHeartbeatStatus,
+					"connection_state": virtualMachine.Summary.Runtime.ConnectionState,
+					"memory": mapstr.M{
+						"overhead":   virtualMachine.Summary.Runtime.MemoryOverhead,
+						"total_size": virtualMachine.Summary.Config.MemorySizeMB,
+						"reserved":   virtualMachine.Summary.Config.MemoryReservation,
+					},
+					"power_state":                   virtualMachine.Summary.Runtime.PowerState,
+					"snapshot_consolidation_needed": virtualMachine.Summary.Runtime.ConsolidationNeeded,
+					"vmx_path":                      virtualMachine.Summary.Config.VmPathName,
 				}
+
+				// add mapped values
+				resourcePool := rpMap[virtualMachine.ResourcePool.Value]
+				event.Put("resource_pool.name", resourcePool.Name)
+				event.Put("resource_pool.path", resourcePool.InventoryPath)
+
+				for _, value := range metric.Value {
+					switch series := value.(type) {
+					case *types.PerfMetricIntSeries:
+						counter := counterMap[series.Id.CounterId]
+						event.Put(counter, series.Value)
+					case *types.PerfMetricSeriesCSV:
+						counter := counterMap[series.Id.CounterId]
+						event.Put(counter, series.Value)
+					default:
+						m.Logger().Debug("Metric is of an unknown type, skipping")
```

Comment:
```suggestion
						m.Logger().Debug("metric is of an unknown type, skipping")
```

In all other places you have use smallcase. So suggesting the change for consistency.

Reply:
updated